rd_ui/app/scripts/controllers/query_source.js

```diff
@@ -70,12 +71,28 @@
       return savePromise;
     };
 
+    $scope.forkQuery = function(options, data) {
+      var savePromise = forkQuery(options, data);
+
+      if (!savePromise) {
+        return;
+      }
+
+      savePromise.then(function(savedQuery) {
+        queryText = savedQuery.query;
+
+        if (isNewQuery) {
```

Comment:
`isNewQuery` will never be true, because you can't fork a new query.

But don't we need to update the `$location.path` in any case, as the query id changes to the forked one?